opensimplex/opensimplex.py

```diff
@@ -19,40 +19,40 @@ def wrapper(func):
 
 # This class is provided for backwards compatibility and might disappear in the future. Use at your own risk.
 class OpenSimplex(object):
-    def __init__(self, seed=DEFAULT_SEED):
+    def __init__(self, seed: int = DEFAULT_SEED) -> None:
         self._perm, self._perm_grad_index3 = _init(seed)
 
-    def noise2(self, x, y):
+    def noise2(self, x: float, y: float) -> float:
         return _noise2(x, y, self._perm)
 
-    def noise2array(self, x, y):
+    def noise2array(self, x: np.ndarray, y: np.ndarray) -> np.ndarray:
         return _noise2a(x, y, self._perm)
 
-    def noise3(self, x, y, z):
+    def noise3(self, x: float, y: float, z: float) -> float:
         return _noise3(x, y, z, self._perm, self._perm_grad_index3)
 
-    def noise3array(self, x, y, z):
+    def noise3array(self, x: np.ndarray, y: np.ndarray, z: np.ndarray) -> np.ndarray:
         return _noise3a(x, y, z, self._perm, self._perm_grad_index3)
 
-    def noise4(self, x, y, z, w):
+    def noise4(self, x: float, y: float, z: float, w: float) -> float:
         return _noise4(x, y, z, w, self._perm)
 
-    def noise4array(self, x, y, z, w):
+    def noise4array(self, x: np.ndarray, y: np.ndarray, z: np.ndarray, w: np.ndarray) -> np.ndarray:
         return _noise4a(x, y, z, w, self._perm)
 
 
 ################################################################################
 
 
-def overflow(x):
+def overflow(x: int) -> int:
     # Since normal python ints and longs can be quite humongous we have to use
     # self hack to make them be able to overflow.
     # Using a np.int64 won't work either, as it will still complain with:
     # "OverflowError: int too big to convert"
     return c_int64(x).value
 
 
-def _init(seed=DEFAULT_SEED):
+def _init(seed: int = DEFAULT_SEED) -> tuple[np.ndarray, np.ndarray]:
```

Comment:
This line fails the tests:

```
=====================================================================
ERROR: Failure: TypeError ('type' object is not subscriptable)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/home/lmas/code/public/opensimplex/devenv/lib/python3.8/site-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/usr/home/lmas/code/public/opensimplex/devenv/lib/python3.8/site-packages/nose/loader.py", line 417, in loadTestsFromName
    module = self.importer.importFromPath(
  File "/usr/home/lmas/code/public/opensimplex/devenv/lib/python3.8/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/home/lmas/code/public/opensimplex/devenv/lib/python3.8/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/usr/local/lib/python3.8/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/local/lib/python3.8/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 702, in _load
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/home/lmas/code/public/opensimplex/tests/test_opensimplex.py", line 11, in <module>
    import opensimplex as simplex
  File "/usr/home/lmas/code/public/opensimplex/opensimplex/__init__.py", line 5, in <module>
    from .opensimplex import OpenSimplex, np
  File "/usr/home/lmas/code/public/opensimplex/opensimplex/opensimplex.py", line 56, in <module>
    def _init(seed: int = DEFAULT_SEED) -> tuple[np.ndarray, np.ndarray]:
TypeError: 'type' object is not subscriptable
```

Reply:
Hmm seems to be a v3.9 thing according to https://docs.python.org/3/library/typing.html#typing.Tuple

Reply:
Been pondering this and I think maybe it's better not to add type hints to the internal functions (in the file `opensimplex.py`)? It's only going to encourage people to use them, when they're not supposed to.
The type hints for the public API in `__init__.py` is still wanted though!

Also don't have to fix this bug with hacky version imports or something (as I want to keep the version compatibility).

Toughs?

Reply:
I think type hinting is not encouraging people to use the functions. After all, what is the problem of people using the internal functions. Moreover, it is better for ourselves to have the code type hinted for finding potential bugs.